pkgs/tools/package-management/nix/tests/compareEval.nix

```diff
@@ -0,0 +1,131 @@
+{
+  fetchFromGitHub,
+  jq,
+  lib,
+  nix,
+  nixVersions,
+  nix-diff,
+  pkgs,
+  pkgs-path ? builtins.fetchGit ../../../../..,
+  runCommand,
+}:
+
+let
+  knownIssues = {
+    # pkgs-lib-formats-tests.drv has floats with different number of digits compared to Nix 2.3
+    "tests.pkgs-lib" = "KNOWN-UNREPRODUCIBLE";
+    # options.json, used in the docs, also a float rendered to string
+    "nixos-install-tools" = "KNOWN-UNREPRODUCIBLE";
+    # same options.json
+    "tests.nixos-functions.nixos-test" = "KNOWN-UNREPRODUCIBLE";
+  };
+
+  evalAllAttributes = nix: mask: import ../../../../test/eval/all-attributes.nix { inherit pkgs nix mask; };
+
+  referenceNix = nixVersions.nix_2_13.overrideAttrs (o: {
```

Comment:
Why this choice of reference Nix? As we do have minVer = 2.3, shouldn't we use Nix 2.3?

Reply:
Nix 2.3 had different logic for printing floats, and it's the only currently packaged Nix that has the old behavior.
By making 2.13 the reference version, we can check all attributes without making any exceptions (`knownIssues`).
Updates of 2.3 will be tested against 2.13, and vice versa (although that is highly unlikely to happen, because both are EOL and 2.13 doesn't receive the attention that 2.3 still does).